src/widget/input/completion.rs

```diff
@@ -164,18 +185,81 @@ impl Completion {
         self.selection = Selection::None;
     }
 
-    pub fn process(&mut self, input: &str) {
-        let Some((head, rest)) = input.split_once('/') else {
+    /// Convert a list of User structs into Completion Entries
+    pub fn with_users(&mut self, users: Vec<User>) -> Self {
+        self.users = users
+            .iter()
+            .map(|u| {
+                // Let's make some eternal strings
+                // TODO! Is this bad?
+                let static_str = Box::leak(u.nickname().to_string().into_boxed_str());
```

Comment:
We definitely don't want to do this! We're leaking this memory here, and we're doing it for every single user nickname every single time the view tree is built!! Memory will grow forever 😬

Instead of trying to use Entry for this, just create a new struct. The args of entry aren't useful here anyways.

Existing Entry can become Command and we can add a Nickname struct, then we can have an Entry enum that wraps both to use for the filtered entries.

Reply:
This has been updated, no more leaky strings.